db/model/subject.js

```diff
@@ -345,8 +349,32 @@ module.exports = function subject(seq, dataTypes) {
           }
         }
 
+        const subjectDepth =
+             inst.absolutePath.split(constants.absolutePathSeparator).length;
+
+        /*
+         * If the depth of the subject does not match the hierarchyLevel
+         * throw an error
+         */
+        if (subjectDepth !== inst.hierarchyLevel) {
+          throw new InvalidHierarchyLevel('The updated depth of ' +
+            `${inst.absolutePath}: ${subjectDepth}, does not match the ` +
+             `updated hierarchyLevel: ${inst.hierarchyLevel}`);
+        }
+
```

Comment:
I'm fine with just using the "read only fields" logic you added in the API layer.

In fact, can you add a user story to extend that paradigm to other models which have read-only fields? e.g. sample "status", "previousStatus", "statusLastChangedAt"...